### PR TITLE
Do no use `x-opaque-id` for deduplicating Elastic originating requests

### PR DESCRIPTION
**Pull request by pgomulka**
deprecated log messages originating from any elastic product requests should not be
deduplicated with the use of x-opaque-id. 
If present, the value of `X-elastic-product-origin` will be used as part of the throttling key.

relates #82810

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java [hunk 1]
```diff
@@ -35,13 +36,14 @@
...
  *
- * It is possible to disable use of `x-opaque-id` as a key with {@link RateLimitingFilter#setUseXOpaqueId(boolean) }
+ * It is possible to disable use of `x-opaque-id` as a key with {@link RateLimitingFilter#setUseXOpaqueIdEnabled(boolean) }
  * @see <a href="https://logging.apache.org/log4j/2.x/manual/filters.htmlf">Log4j2 Filters</a>
  */
 @Plugin(name = "RateLimitingFilter", category = Node.CATEGORY, elementType = Filter.ELEMENT_TYPE)
 public class RateLimitingFilter extends AbstractFilter {
-
-    private volatile boolean useXOpaqueId = true;
+    private static final String KIBANA_OPRDUCT_ORIGIN = "kibana";
```
Comment (TinaHeiligers) [inline server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java:44]: is this a typo or intentional?
```suggestion
    private static final String KIBANA_PRODUCT_ORIGIN = "kibana";
```

Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java]: that was a typo. The constant is not needed as this change will apply to all elastic originating requests

File: server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java [hunk 2]
```diff
@@ -34,14 +36,15 @@
...
  *
- * It is possible to disable use of `x-opaque-id` as a key with {@link RateLimitingFilter#setUseXOpaqueId(boolean) }
  * @see <a href="https://logging.apache.org/log4j/2.x/manual/filters.htmlf">Log4j2 Filters</a>
  */
 @Plugin(name = "RateLimitingFilter", category = Node.CATEGORY, elementType = Filter.ELEMENT_TYPE)
 public class RateLimitingFilter extends AbstractFilter {
-
-    private volatile boolean useXOpaqueId = true;
+    // a flag to disable/enable use of xOpaqueId controlled by changing cluster setting
+    private volatile boolean useXOpaqueIdEnabled = true;
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java:47]: Why the addition of "enabled"? The "use" prefix already denotes a boolean.

**Review by rjernst — APPROVED** [review]
LGTM, but I would not change the naming of the member or setter.

File: server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java (hunk 2 above)
Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/common/logging/RateLimitingFilter.java:47]: true - it does not make sens to me now either. I can't remember what I was thinking when renaming..

**Pull request merged**